src/network.toit

```diff
@@ -49,13 +54,23 @@ class EndpointHttp implements Endpoint:
       // firmware if requested to do so.
       validate-firmware --reason="connected to network"
 
+      request-mutex := monitor.Mutex
+
       // We run two tasks concurrently: One broadcasts the device identity
       // via UDP and one serves incoming HTTP requests. We run the tasks
       // in a group so if one of them terminates, we take the other one down
       // and clean up nicely.
       Task.group --required=1 [
         :: broadcast-identity network device address,
-        :: serve-incoming-requests socket device address,
+        :: serve-incoming-requests socket device address request-mutex,
+        // If the call to the network-manager monitor returns, it will terminate the
+        // task and thus shut down the whole group.
+        ::
+          network-manager.wait-for-request-to-disable-network
```

Comment:
Not entirely sure about this approach. It feels like the network manager knows about the endpoints, so maybe it could terminate the endpoint by notifying it somehow and let `serve-incoming-requests` react to that more explicitly?

It feels odd that you're adding another task and then give yourself the synchronization issue where you only want to handle the shutdown at specific points in `serve-incoming-requests`.